public/app/features/alerting/unified/plugins/useRulePluginLinkExtensions.ts

```diff
@@ -0,0 +1,79 @@
+import { PluginExtensionPoints } from '@grafana/data';
+import { getPluginLinkExtensions } from '@grafana/runtime';
+import { CombinedRule } from 'app/types/unified-alerting';
+import { PromRuleType } from 'app/types/unified-alerting-dto';
+
+import { getRuleOrigin } from '../utils/rules';
+
+interface AlertingRuleExtensionContext {
+  name: string;
+  namespace: string;
+  group: string;
+  expression: string;
+  labels: Record<string, string>;
+  annotations: Record<string, string>;
+}
+
+interface RecordingRuleExtensionContext {
+  name: string;
+  namespace: string;
+  group: string;
+  expression: string;
+  labels: Record<string, string>;
+}
+
+export function useRulePluginLinkExtension(rule: CombinedRule) {
+  const ruleOrigin = getRuleOrigin(rule);
+  const ruleType = rule.promRule?.type;
+  if (!ruleOrigin || !ruleType) {
+    return [];
+  }
+
+  const { pluginId } = ruleOrigin;
+
+  switch (ruleType) {
+    case PromRuleType.Alerting:
+      return getAlertingRuleActionPluginLinkExtension({ pluginId, rule });
+    case PromRuleType.Recording:
+      return getRecordingRulePluginLinkExtension({ pluginId, rule });
+    default:
+      return [];
+  }
+}
+
+function getAlertingRuleActionPluginLinkExtension({ pluginId, rule }: { pluginId: string; rule: CombinedRule }) {
```

Comment:
The two methods are very similar (other than `annotations` from what I can see?). Could we generalise it/reuse across them?